orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -350,10 +232,12 @@ def to(self, *args, **kwargs):
     def eval(self: T) -> T:
         self._is_training = False
         self._original_module.eval()
+        self._flattened_output_module.eval()
```

Comment:
If you call `eval()` or `training()` on the most specialized class (`self._flattened_output_module`), all others recursively get updated

```python
import torch

class ORTModule(torch.nn.Module):
    def __init__(self, module):
        super(ORTModule, self).__init__()
        self.orig=module
        self.flat=FlattenedOutputModule(module)

class FlattenedOutputModule(torch.nn.Module):
    def __init__(self, module):
        super(FlattenedOutputModule, self).__init__()
        self._base_module = module


>>> import torch
>>>
>>> class ORTModule(torch.nn.Module):
...     def __init__(self, module):
...         super(ORTModule, self).__init__()
...         self.orig=module
...         self.flat=FlattenedOutputModule(module)
...
>>> class FlattenedOutputModule(torch.nn.Module):
...     def __init__(self, module):
...         super(FlattenedOutputModule, self).__init__()
...         self._base_module = module
...
>>> model
Linear(in_features=10, out_features=1, bias=True)
>>> model1=ORTModule(model)
>>> model1
ORTModule(
  (orig): Linear(in_features=10, out_features=1, bias=True)
  (flat): FlattenedOutputModule(
    (_base_module): Linear(in_features=10, out_features=1, bias=True)
  )
)
>>> model1.training
True
>>> model1.flat.training
True
>>> model1.flat._base_module.training
True
>>> model1.eval()
ORTModule(
  (orig): Linear(in_features=10, out_features=1, bias=True)
  (flat): FlattenedOutputModule(
    (_base_module): Linear(in_features=10, out_features=1, bias=True)
  )
)
>>> model1.flat._base_module.training
False
>>> model1.flat.training
False
>>> model1.training
False
```

Reply:
I can probably remove the original module eval and train calls.